bin/validator_gen_php.py

```diff
@@ -0,0 +1,399 @@
+
+  This method reads the specfile and emits PHP to sys.stdout.
+
+  Args:
+    specfile: Path to validator.protoascii, the specfile to generate
+        PHP from.
+    validator_pb2: The proto2 Python module generated from validator.proto.
+    text_format: The text_format module from the protobuf package, e.g.
+        google.protobuf.text_format.
+    descriptor: The descriptor module from the protobuf package, e.g.
+        google.protobuf.descriptor.
+    out: a list of lines to output (without the newline characters), to
+        which this function will append.
+  """
+  # First, find the descriptors and enums and generate PHP
+  # classes and enums.
+  msg_desc_by_name = {}
+  enum_desc_by_name = {}
+  FindDescriptors(validator_pb2, msg_desc_by_name, enum_desc_by_name)
+
+  rules_obj = '%s.RULES' % validator_pb2.DESCRIPTOR.package
+  all_names = [rules_obj] + msg_desc_by_name.keys() + enum_desc_by_name.keys()
+  all_names.sort()
+
+  output_header = """<?php
+/*
+ * Copyright 2016 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Generated by validator_gen_php.py - do not edit.
+//   validator_gen_php.py is available in a Lullabot forked repo https://github.com/Lullabot/amphtml
+//   Be sure to look in the `php-validator-generated` branch of the repo in the `validator` folder
+//
+
+namespace Lullabot\AMP\Spec;
+"""
+
+  out.append(output_header);
+
+  for name in all_names:
+    if name in msg_desc_by_name:
+      PrintClassFor(descriptor, msg_desc_by_name[name], out)
+    elif name in enum_desc_by_name:
+      PrintEnumFor(enum_desc_by_name[name], out)
+
+  # Read the rules file, validator.protoascii by parsing it as a text
+  # message of type ValidatorRules.
+  rules = validator_pb2.ValidatorRules()
+  text_format.Merge(open(specfile).read(), rules)
+  out.append('class ValidationRulesFactory {')
+  out.append('  public static function createValidationRules() {')
+  PrintObject(descriptor, rules, 0, out)
+  out.append('  return $o_0;')
+  out.append('  }')
+  out.append('}')
+  out.append('')
+
+def Main():
+	"""The main method, which executes all build steps and runs the tests."""
+	logging.basicConfig(
+			format='[[%(filename)s %(funcName)s]] - %(message)s', level=logging.INFO)
+
+	out_dir = 'dist'
+
+	SetupOutDir(out_dir)
+	GenValidatorProtoascii(out_dir)
+	GenValidatorPb2Py(out_dir)
+	GenValidatorProtoascii(out_dir)
+	GeneratePHP(out_dir)
+
+if __name__ == '__main__':
+	Main()
```

Comment:
Adapted version of https://github.com/Lullabot/amphtml/blob/php-validator-generated/validator/validator_gen_php.py
This file can run as standalone